core/common/src/main/java/alluxio/util/CommonUtils.java

```diff
@@ -717,5 +717,27 @@ private synchronized T firstTime() {
     };
   }
 
+  /**
+   * Partitions a list into numLists many lists each with around list.size() / numLists elements.
+   *
+   * @param list the list to partition
+   * @param numLists number of lists to return
+   * @param <T> the object type
+   * @return partitioned list
+   */
+  public static <T> List<List<T>> partition(List<T> list, int numLists) {
+    ArrayList<List<T>> result = new ArrayList<>(numLists);
+
+    for (int i = 0; i < numLists; i++) {
+      result.add(new ArrayList<>(list.size() / numLists + 1));
```

Comment:
Will this be a problem if `list.size` is smaller than `numLists + 1`?

Reply:
1. divide happens before + 1 so it's guaranteed to be at least 1 (as long as numLists is > 0).
2. But more importantly, new ArrayList<>(0) is fine because it's just initial capacity.